### PR TITLE
 Bug 1946479: inject SA tokens as projected volumes to KAS install and prune pods

### PR DESCRIPTION
**Pull request by stlaz**
I am not sure this will work but I at least want to see it fail and I don't have a cluster handy.

### CODE REVIEW

**Comment by s-urbaniak** [comment]
xref https://github.com/openshift/kubernetes/pull/714

File: bindata/v4.1.0/kube-apiserver/pod.yaml [hunk 1]
```diff
@@ -243,3 +259,21 @@ spec:
   - hostPath:
       path: /var/log/kube-apiserver
     name: audit-dir
+  - name: kube-api-access
+    projected:
```
Comment (s-urbaniak) [inline bindata/v4.1.0/kube-apiserver/pod.yaml:263]: i think the keyword `projected` is wrong here :thinking:

**Comment by stlaz** [comment]
/test e2e-upgrade
infra

**Comment by stlaz** [comment]
/retest
just for the heck of it, but this is very likely broken, somehow

**Comment by stlaz** [comment]
/retest
seems irrelevant

**Comment by stlaz** [comment]
/retest
no quota

**Comment by stlaz** [comment]
Another PR merged, changing lib-go to an older version. Rebased.

**Comment by mfojtik** [comment]
/override ci/prow/e2e-aws
/override ci/prow/e2e-aws-serial
/override ci/prow/e2e-gcp-operator
/override ci/prow/e2e-upgrade 
/override ci/prow/k8s-e2e-gcp

Overriding these as the k8s-e2e-gcp failure is unrelated to this change and this PR is blocking few other PR's required to close 4.8 blocker.

**Pull request merged**